torch/optim/adagrad.py

```diff
@@ -50,7 +50,7 @@ def __init__(
         for group in self.param_groups:
             for p in group["params"]:
                 state = self.state[p]
-                state["step"] = torch.tensor(0.0)
+                state["step"] = torch.tensor(0.0, dtype=torch.float)
```

Comment:
for more explicitness, should this use torch.float32 (or torch.float64?) instead? even in docs, it seems very difficult/impossible to find what torch.float stands for: https://pytorch.org/docs/stable/search.html?q=torch.float&check_keywords=yes&area=default

Reply:
it should be torch.float32, float is 32 and double is 64

Reply:
Sure, I can change it to float32 to be clearer. But yes, the float/double naming convention comes from C/C++

Reply:
done